analyzer/codechecker_analyzer/analyzers/config_handler.py

```diff
@@ -129,32 +130,64 @@ def __gen_name_variations(self):
         return reserved_names
 
     def initialize_checkers(self,
-                            available_profiles,
-                            package_root,
+                            analyzer_context,
                             checkers,
-                            checker_config=None,
-                            cmdline_checkers=None,
+                            cmdline_enable=[],
                             enable_all=False):
         """
-        Initializes the checker list for the specified config handler based on
-        given checker profiles, commandline arguments and the
-        analyzer-retrieved checker list.
+        Add checkers and set their "enabled/disabled" status. The following
+        inputs are considered in this order:
+        - First the default state is taken based on the analyzer tool.
+        - Members of "default" profile are enabled.
+        - In case of "--enable-all" every checker is enabled except for "alpha"
+          and "debug" checker groups. "osx" checker group is also not included
+          unless the target platform is Darwin.
+        - Command line "--enable/--disable" flags.
+          - Their arguments may start with "profile:" or "guideline:" prefix
+            which makes the choice explicit.
+          - Without prefix it means a profile name, a guideline name or a
+            checker group/name in this priority order.
```

Comment:
I am thinking to put this section into our user guide too because this can be important for the users. What do you think @bruntib?

Reply:
It is there at the documentation of `--enable`/`--disable` flags.

Reply:
There is an extra checkers and profiles related section in the user guide: https://github.com/Ericsson/codechecker/blob/master/docs/analyzer/user_guide.md#toggling-checkers